QgisModelBaker/gui/workflow_wizard/execution_page.py

```diff
@@ -75,6 +75,21 @@ def __init__(self, parent, title, db_action_type):
         self.is_complete = False
         self.pending_sessions = []
 
+    def cancel_sessions(self):
```

Comment:
@domi4484 Hey Mr. QTimer ;-) Any idea what I am doing wrong here?
Because I don't receive the signal `run_finished` before the timer expires. It's kind of blocked as long as the loop is executing...


And in general. What do you think about this approach with blocking `reject()` until the QProcesses are terminated?

Reply:
It is blocked because (I suppose) cancel_session is called by the main thread and the call to `loop.exec()` will block the main thread where the `timer` was created. Why are you unsing an event loop here? Just to block until canceling or timeout?
There are also other ways to wait without blocking the main thread

I think it is fine to block in the reject function as long as you display a hourglass to tell the user you are not hanging but doing a long operation.

Reply:
Thanks. Yes I assume something like that. But the main thread should be blocked in fact. But be released on receiving `run_finished`. We use that on other places as well, but I think there we download a file with QNetworkRequest where it possibly creates a second thread...